tests/components/insteon/test_config_flow.py

```diff
@@ -0,0 +1,703 @@
+        result = await hass.config_entries.options.async_init(entry_id)
+
+    assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+    assert result["step_id"] == "init"
+
+    result2 = await hass.config_entries.options.async_configure(
+        result["flow_id"], {step: True},
+    )
+    return result2
+
+
+async def _options_form(hass, flow_id, user_input):
+    """Test an options form."""
+
+    with patch(PATCH_ASYNC_SETUP_ENTRY, return_value=True) as mock_setup_entry:
+        result = await hass.config_entries.options.async_configure(flow_id, user_input)
+        return result, mock_setup_entry
+
+
+async def test_options_change_hub_config(hass: HomeAssistantType):
+    """Test changing Hub v2 config."""
+    config_entry = MockConfigEntry(
+        domain=DOMAIN,
+        entry_id="abcde12345",
+        data={**MOCK_USER_INPUT_HUB_V2, CONF_HUB_VERSION: 2},
+        options={},
+    )
+
+    config_entry.add_to_hass(hass)
+    result = await _options_init_form(
+        hass, config_entry.entry_id, STEP_CHANGE_HUB_CONFIG
+    )
+
+    user_input = {
+        CONF_HOST: "2.3.4.5",
+        CONF_PORT: 9999,
+        CONF_USERNAME: "new username",
+        CONF_PASSWORD: "new password",
+    }
+    result, _ = await _options_form(hass, result["flow_id"], user_input)
+
+    assert result["type"] == data_entry_flow.RESULT_TYPE_CREATE_ENTRY
+    assert config_entry.options == {}
+    assert config_entry.data == {**user_input, CONF_HUB_VERSION: 2}
+
+
+async def test_options_add_device_override(hass: HomeAssistantType):
+    """Test adding a device override."""
+    config_entry = MockConfigEntry(
+        domain=DOMAIN,
+        entry_id="abcde12345",
+        data={**MOCK_USER_INPUT_HUB_V2, CONF_HUB_VERSION: 2},
+        options={},
+    )
+
+    config_entry.add_to_hass(hass)
+    result = await _options_init_form(hass, config_entry.entry_id, STEP_ADD_OVERRIDE)
+
+    user_input = {
+        CONF_ADDRESS: "1a2b3c",
+        CONF_CAT: "0x04",
+        CONF_SUBCAT: "0xaa",
+    }
+    result, _ = await _options_form(hass, result["flow_id"], user_input)
+
+    assert result["type"] == data_entry_flow.RESULT_TYPE_CREATE_ENTRY
+    assert len(config_entry.options[CONF_OVERRIDE]) == 1
+    assert config_entry.options[CONF_OVERRIDE][0][CONF_ADDRESS] == "1A.2B.3C"
+    assert config_entry.options[CONF_OVERRIDE][0][CONF_CAT] == 4
+    assert config_entry.options[CONF_OVERRIDE][0][CONF_SUBCAT] == 170
+
+    result2 = await _options_init_form(hass, config_entry.entry_id, STEP_ADD_OVERRIDE)
+
+    user_input = {
+        CONF_ADDRESS: "4d5e6f",
+        CONF_CAT: "05",
+        CONF_SUBCAT: "bb",
+    }
+    await _options_form(hass, result2["flow_id"], user_input)
+
+    assert len(config_entry.options[CONF_OVERRIDE]) == 2
+    assert config_entry.options[CONF_OVERRIDE][1][CONF_ADDRESS] == "4D.5E.6F"
+    assert config_entry.options[CONF_OVERRIDE][1][CONF_CAT] == 5
+    assert config_entry.options[CONF_OVERRIDE][1][CONF_SUBCAT] == 187
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5985897)